pkg/checks/checks.go

```diff
@@ -54,68 +56,182 @@ func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urlto
 	}, ctx.Done())
 }
 
-func checkNamespace(namespace string) bool {
-	if namespace == "default" {
-		return true
-	}
+type Node struct {
+	Ready bool
+	Name  string
+}
 
-	whitelistedNamespaces := []string{
-		"openshift-",
-		"kube-",
-	}
-	for _, ns := range whitelistedNamespaces {
-		if strings.HasPrefix(namespace, ns) {
-			return true
+func GenerateNodeMap(cs *acsapi.OpenShiftManagedCluster) map[string][]Node {
+	nodes := make(map[string][]Node)
+	for _, app := range cs.Properties.AgentPoolProfiles {
+		nodes[string(app.Role)] = []Node{}
+		for i := 0; i < app.Count; i++ {
+			n := Node{Name: fmt.Sprintf("%s-%06d", app.Role, i)}
+			nodes[string(app.Role)] = append(nodes[string(app.Role)], n)
 		}
 	}
-	return false
+	return nodes
 }
 
-// WaitForInfraServices verify daemonsets, statefulsets
-func WaitForInfraServices(ctx context.Context, appclient *appclient.AppsV1Client) error {
-	log.Info("checking infrastructure service health")
-out:
-	for {
-		_, err := appclient.Deployments("openshift-web-console").Get("webconsole", metav1.GetOptions{})
-		switch {
-		case err == nil:
-			break out
-		case kerrors.IsNotFound(err):
-		default:
-			return err
-		}
-		select {
-		case <-time.After(2 * time.Second):
-		case <-ctx.Done():
-			return ctx.Err()
+func WaitForNodeReady(ctx context.Context, nodes map[string][]Node, kc *kubernetes.Clientset) error {
+	for nodeType, ntNodes := range nodes { // walk node types, e.g. master, compute, infra
+		for idx, node := range ntNodes { // walk each node in type, e.g. master00000[1,2,3]
+		nodeloop:
+			for {
+				n, err := kc.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+				if err == nil {
+					for _, cond := range n.Status.Conditions {
```

Comment:
This loop can be abstracted to a `IsNodeReady(node) bool` helper. Then you shouldn't need the `nodeloop` label I think.

Reply:
That makes a lot of sense.  Let me move it.